src/System/local_consistency.rs

```diff
@@ -1,128 +1,42 @@
-    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        match self {
-            ConsistencyFailure::NoInitialLocation => write!(f, "No Initial State"),
-            ConsistencyFailure::EmptyInitialState => write!(f, "Empty Initial State"),
-            ConsistencyFailure::NotConsistentFrom(location, action) => {
-                write!(
-                    f,
-                    "Not Consistent From {} Failing action {}",
-                    location, action
-                )
-            }
-            ConsistencyFailure::NotDeterministicFrom(location, action) => {
-                write!(
-                    f,
-                    "Not Deterministic From {} Failing action {}",
-                    location, action
-                )
-            }
-            ConsistencyFailure::NotDisjoint(srf) => {
-                write!(f, "Not Disjoint: {} {:?}", srf.reason, srf.actions)
-            }
-        }
-    }
-}
-
-/// The result of a determinism check.
-/// Failure includes the [LocationID].
-pub enum DeterminismResult {
-    Success,
-    //Failure should contain a DeterminsmFailure which should contain either a location + a string, or a SystemRecipeFailure
-    Failure(DeterminismFailure),
-}
-
-pub enum DeterminismFailure {
-    NotDeterministicFrom(LocationID, String),
-    NotDisjoint(SystemRecipeFailure),
-}
-
-impl fmt::Display for DeterminismResult {
-    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        match self {
-            DeterminismResult::Success => write!(f, "Success"),
-            DeterminismResult::Failure(DeterminismFailure::NotDeterministicFrom(
-                location,
-                action,
-            )) => {
-                write!(
-                    f,
-                    "Not Deterministic From {} failing action {}",
-                    location, action
-                )
-            }
-            DeterminismResult::Failure(DeterminismFailure::NotDisjoint(srf)) => {
-                write!(f, "Not Disjoint: {} {:?}", srf.reason, srf.actions)
-            }
-        }
-    }
-}
+use super::query_failures::{ConsistencyResult, DeterminismResult};
 
 ///Local consistency check WITH pruning.
 pub fn is_least_consistent(system: &dyn TransitionSystem) -> ConsistencyResult {
     if system.get_initial_location().is_none() {
-        return ConsistencyResult::Failure(ConsistencyFailure::NoInitialLocation);
+        return ConsistencyFailure::no_initial_state(system);
         //TODO: figure out whether we want empty TS to be consistent
     }
 
     let mut passed = vec![];
     let state = system.get_initial_state();
     if state.is_none() {
         warn!("Empty initial state");
-        return ConsistencyResult::Failure(ConsistencyFailure::EmptyInitialState);
+        return ConsistencyFailure::no_initial_state(system);
     }
     let mut state = state.unwrap();
     state.extrapolate_max_bounds(system);
     consistency_least_helper(state, &mut passed, system)
```

Comment:
I think this can be simplified:
```suggestion
    if system.get_initial_location().is_none() {
        ConsistencyFailure::no_initial_state(system)
        //TODO: figure out whether we want empty TS to be consistent
    } else if let Some(mut state) = system.get_initial_state() {
        let mut passed = vec![];
        state.extrapolate_max_bounds(system);
        consistency_least_helper(state, &mut passed, system)
    } else {
        ConsistencyFailure::no_initial_state(system)
    }
```